pkg/resources/citadel/deployment.go

```diff
@@ -77,6 +77,13 @@ func (r *Reconciler) deployment() runtime.Object {
 		)
 	}
 
+	if r.Config.Spec.TrustDomain != "" {
+		args = append(args,
```

Comment:
Could not we use this setting with a previously set default `cluster.local` without the condition?

Reply:
Probably, I just didn't want to go from implicit to explicit without a discussion about it